File: Dataset/Leetcode/test/101/78.java
 class Solution {
    public boolean XXX(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<TreeNode>();
        queue.offer(root.left);
        queue.offer(root.right);
        while(!queue.isEmpty()){
            TreeNode node1 = queue.poll();
            TreeNode node2 = queue.poll();
            if(node1 == null && node2 == null) continue;
            if(node1 == null || node2 == null) return false;
            if(node1.val != node2.val) return false;
            queue.offer(node1.left);
            queue.offer(node2.right);
            queue.offer(node1.right);
            queue.offer(node2.left);
        }
        return true;
    }
}

File: Dataset/Leetcode/valid/20/78.py
 class Solution:
    def XXX(self, s: str) -> bool:
        li = []
        for i in s:
            if i == '(' or i == '[' or i == '{':
                li.append(i)
            if i == ')' or i == ']' or i == '}':
                if li == []:
                    return False
                elif abs(ord(i) - ord(li.pop(-1))) > 2:
                    return False
        return li == []

